shakedown/dcos/package.py

```diff
@@ -51,7 +51,8 @@ def install_package(
         options_json=None,
         wait_for_completion=False,
         timeout_sec=600,
-        expected_running_tasks=0
+        expected_running_tasks=0,
+        wait_for_marathon_deployment=True
```

Comment:
This addition feels like a usability regression.

* this method is pretty confused. There is `install_package` and `install_package_and_wait` yet the former is the one that does the waiting and the latter is a meaningless thin wrapper (it only differs in having a different default keyword argument).  If anything, we should move any blocking/wait related logic to that function and keep this function strictly non-blocking or kill the other function
* competing args: There are now 4 different keyword arguments that control blocking (wait_for_completion, expected_running_tasks, wait_for_marathon_deployment) Interdependent args make it very easy to accidentally make a call which does not have the intended effects. E.G. before, `wait_for_completion=True` guaranteed that this function would block. Now, `wait_for_completion=True` will do nothing at all if `wait_for_marathon_deployment=False` and service_name and expected_running_tasks are left as default
* Use-case confusion: this pre-dates this PR, but why does this method do two very specific kinds of waiting? First it waits for the number of expected tasks to be running and then it checks the deployment state. Wouldnt the latter necessarily include the former unless the former check was invalid?
* buried if statements to control high-level behavior create testing/validation issues. I.E. each toggle that is added like this means we really have two separate functions to test

TL;DR This method has kind of already gone off the rails. Rather, we should go into specific test suites/cases and change the usage of this method to something of the pattern: install_package -> retry checking finish criteria. Where the former is extremely minimal (it ONLY installs) and the later is a function paired with a retrying.retry decorator